### PR TITLE
vmalert: add command line flag `-notifier.headers`

### PR DESCRIPTION
**Pull request by Haleygo**
to allow configuring additional headers in each request to the corresponding notifier.
Other flags like `-datasource.headers`, `-remoteWrite.headers` already use `^^` as delimiter, it's consistent to use it in `-notifier.headers` as well.

related https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3260
vmalert can integrate with alertmanager that supports multi-tenant by adding tenantID header`X-Scope-OrgID` in requests.
In multitenancy, vmalert can also filter alerts which send to different notifier addresses(or with different header settings) using  `alert_relabel_configs`.

### CODE REVIEW

File: app/vmalert/notifier/init.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ var (
 		"Enable this flag if you want vmalert to evaluate alerting rules without sending any notifications to external receivers (eg. alertmanager). "+
 		"-notifier.url, -notifier.config and -notifier.blackhole are mutually exclusive.")
 
+	headers = flagutil.NewArrayString("notifier.headers", "Optional HTTP headers to send with each request to the corresponding -notifier.url. "+
```
Comment (hagen1778) [inline app/vmalert/notifier/init.go:28]: shall we mention that this flag will be respected only if `notifier.url` is specified?

Comment (Haleygo) [reply app/vmalert/notifier/init.go:28]: It's mentioned in the description `xxxx to the corresponding -notifier.url`, as other flags like `-notifier.basicAuth.username`.

**Comment by Haleygo** [comment]
Split the bugfix part of `notifier.config` to seperate PR https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6762, so we can backfill it to LTS.

**Comment by hagen1778** [comment]
@Haleygo could you please rebase this PR on `master` branch?

**Comment by Haleygo** [comment]
> @Haleygo could you please rebase this PR on `master` branch?

updated

**Comment by Haleygo** [comment]
> Hey @Haleygo. It looks to me like this PR contains 2 changes:
> 
> 1. FEATURE: support of `-notifier.headers`
> 2. BUGFIX: fix for relabeling for alerting notifications
>    These changes seems orthogonal, and 2nd change isn't mentioned in changelog. I think we should split this PR into 2 PRs with the corresponding changes.

Splitted bugfix to https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6823, please check.

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Pull request merged**